dotrix_core/src/renderer/color.rs

```diff
@@ -0,0 +1,132 @@
+use std::ops::{Index, IndexMut};
+
+/// RGBA Color.
+#[derive(Copy, Clone, Debug)]
+pub struct Color {
+    /// Red channel. Should be in range from 0 to 1.
+    pub r: f32,
+    /// Green channel. Should be in range from 0 to 1.
+    pub g: f32,
+    /// Blue channel. Should be in range from 0 to 1.
+    pub b: f32,
+    /// Alpha channel. Should be in range from 0 to 1.
+    pub a: f32,
+}
+
+impl Color {
+    /// RGB Constructor, values should be in range from 0 to 1. Alpha will be 1.
+    pub fn rgb(r: f32, g: f32, b: f32) -> Self {
+        Self { r, g, b, a: 1.0 }
+    }
+
+    /// RGBA Constructor. Values should be in range from 0 to 1.
+    pub fn rgba(r: f32, g: f32, b: f32, a: f32) -> Self {
+        Self { r, g, b, a }
+    }
+
+    /// Red color (r: 1.0, g: 0.0, b: 0.0)
+    pub fn red() -> Self {
+        Self::rgb(1.0, 0.0, 0.0)
+    }
+
+    /// Green color (r: 0.0, g: 1.0, b: 0.0)
+    pub fn green() -> Self {
+        Self::rgb(0.0, 1.0, 0.0)
+    }
+
+    /// Blue color (r: 0.0, g: 0.0, b: 1.0)
+    pub fn blue() -> Self {
+        Self::rgb(0.0, 0.0, 1.0)
+    }
+
+    /// Cyan color (r: 0.0, g: 1.0, b: 1.0)
+    pub fn cyan() -> Self {
+        Self::rgb(0.0, 1.0, 1.0)
+    }
+
+    /// Magenta color (r: 1.0, g: 0.0, b: 1.0)
+    pub fn magenta() -> Self {
+        Self::rgb(1.0, 0.0, 1.0)
+    }
+
+    /// Yellow color (r: 1.0, g: 1.0, b: 0.0)
+    pub fn yellow() -> Self {
+        Self::rgb(1.0, 1.0, 0.0)
+    }
+
+    /// White color (r: 1.0, g: 1.0, b: 1.0)
+    pub fn white() -> Self {
+        Self::rgb(1.0, 1.0, 1.0)
+    }
+
+    /// Black color (r: 0.0, g: 0.0, b: 0.0)
+    pub fn black() -> Self {
+        Self::rgb(0.0, 0.0, 0.0)
+    }
+
+    /// Orange color (r: 1.0, g: 0.5, b: 0.0)
+    pub fn orange() -> Self {
+        Self::rgb(1.0, 0.5, 0.0)
+    }
+
+    /// Multiply Color by f32. Result is not clamped.
+    pub fn mul_f32(self, rhs: f32) -> Self {
+        Self {
+            r: self.r * rhs,
+            g: self.g * rhs,
+            b: self.b * rhs,
+            a: self.a * rhs,
+        }
+    }
+
+    /// Create color from array
+    pub fn from_f32_3(rgb: [f32; 3]) -> Self {
```

Comment:
Consider use of `From` trait
https://doc.rust-lang.org/std/convert/trait.From.html

Reply:
Implemented, is there any benefit to use rather than `from_f32_3`? Have them both for now

Reply:
This is a standard practice with Rust. Benefit is that if you know that you can construct from a slice and let say vector, you can use the same function for both cases: `Color::from(vector)` and `Color::from(slice)`.